tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/command/CommandFulfiller.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.teradata.tempto.internal.fulfillment.command;
+
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.context.State;
+import com.teradata.tempto.fulfillment.RequirementFulfiller;
+import com.teradata.tempto.fulfillment.TestStatus;
+import com.teradata.tempto.fulfillment.command.Command;
+import com.teradata.tempto.fulfillment.command.CommandRequirement;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public abstract class CommandFulfiller<T extends CommandRequirement>
+        implements RequirementFulfiller
+{
+    private static final Logger LOGGER = getLogger(CommandFulfiller.class);
+
+    private final Class<T> requirementClass;
+
+    CommandFulfiller(Class<T> requirementClass)
+    {
+        this.requirementClass = requirementClass;
+    }
+
+    @Override
+    public final Set<State> fulfill(Set<Requirement> requirements)
+    {
+        requirements.stream()
+                .filter(requirement -> requirement.getClass().isAssignableFrom(requirementClass))
+                .map(requirement -> (T) requirement)
+                .distinct()
+                .forEach(this::fulfill);
+
+        return emptySet();
+    }
+
+    private void fulfill(T requirement) {
+        requirement.getSetupCommands().stream()
+                .map(Command::getCommand)
+                .forEach(CommandFulfiller::execute);
+    }
+
+    private static void execute(String command)
+    {
+        try {
+            LOGGER.info("Executing command: " + command);
+            Process process = Runtime.getRuntime().exec(command);
+            int exitStatus = process.waitFor();
```

Comment:
I think we need to read stdout and stderr of command here. Otherwise it can never exit.